File: src/main/java/com/reelcosmos/service/auth/AuthService.java
package com.reelcosmos.service.auth;

import com.reelcosmos.dto.auth.LoginRequest;
import com.reelcosmos.dto.auth.RefreshTokenRequest;
import com.reelcosmos.dto.auth.RegisterRequest;
import com.reelcosmos.dto.response.JwtResponse;

public interface AuthService {

    JwtResponse register(RegisterRequest request);

    JwtResponse login(LoginRequest request);

    JwtResponse refreshToken(RefreshTokenRequest request);

    void logout();
}

File: frontend/src/services/authService.ts
import api from "../api/axios";

import type {
    LoginRequest,
    RegisterRequest,
    JwtResponse,
} from "../types/auth";

const AUTH_URL = "/api/auth";

export const login = async (
    data: LoginRequest
): Promise<JwtResponse> => {
    const response = await api.post<JwtResponse>(
        `${AUTH_URL}/login`,
        data
    );

    return response.data;
};

export const register = async (
    data: RegisterRequest
): Promise<JwtResponse> => {
    const response = await api.post<JwtResponse>(
        `${AUTH_URL}/register`,
        data
    );

    return response.data;
};

export const logout = async (): Promise<void> => {
    await api.post(`${AUTH_URL}/logout`);
};

export const refreshToken = async (
    refreshToken: string
): Promise<JwtResponse> => {
    const response = await api.post<JwtResponse>(
        `${AUTH_URL}/refresh`,
        {
            refreshToken,
        }
    );

    return response.data;
};